File: exampleSources/swift/app2.swift
// Swift — protocol + extension + conformance.

protocol Describable {
    var summary: String { get }
}

struct Task: Describable {
    var title: String
    var priority: Int
}

extension Task {
    var summary: String { "\(priority)  \(title)" }
}

extension Sequence where Element == Task {
    var titles: [String] { map(\.title) }
}

let backlog: [Task] = [
    .init(title: "Write tests",     priority: 2),
    .init(title: "Fix login bug",   priority: 5),
    .init(title: "Refactor parser", priority: 3),
]

for task in backlog {
    print(task.summary)
}
print("count = \(backlog.titles.count)")

File: exampleSources/kotlin/app2.kt
// Kotlin — sealed class + when expression over a small state.

sealed class Status {
    object Open : Status()
    object Done : Status()
    object Blocked : Status()
}

data class Task(val title: String, val priority: Int, val status: Status)

fun describe(t: Task): String = when (t.status) {
    Status.Open    -> "TODO  ${t.title} (pri=${t.priority})"
    Status.Done    -> "OK   ${t.title}"
    Status.Blocked -> "WAIT ${t.title}"
}

fun main() {
    val backlog = listOf(
        Task("Write tests", 2, Status.Open),
        Task("Fix login bug", 5, Status.Blocked),
        Task("Refactor parser", 3, Status.Done),
    )
    backlog.forEach { println(describe(it)) }
}